WordPress/openverse-frontend
# Use links instead of buttons for header search type switcher

## Fixes


Fixes #1130 by @obulat

**Note**: This PR works for adding links that retain common filter query parameters, but for correctly handling the search type specific query parameters it is dependent on #1040 (it needs an action that is only available in that PR). Since these two PRs are interdependent, it is best to approve and merge them together, or add `TODO` into another PR once one of them is merged.

## Description


This PR makes the search type switcher use `VLink` when used in the header, and keep the current button status when used in the homepage searchbar.
This makes the header links more semantic (and also sets the `aria-current="page"` for the link to currently selected search type), and simplifies the search type handling.
It also fixes the e2e tests that are failing in #1040, and are currently disabled.

## Testing Instructions

Run the app using `pnpm dev`, search for something from the homepage.

Switching search type from the homepage search bar should work as usual, and should not redirect to another page, because it's not using an `<a>` link.

Selecting a search type from the header should redirect you, as usual, and update the url path. You can see in the devtools that the search type switcher is using `<a>` links. 

Currently, the link will only retain the filter query parameters that are common for all search types (license, license types, search by creator).
After the search store change is merged, You should also see that the query in those links retains the filter parameters that that type can use, and removes the parameters that are not appropriate. For example, if on the image page you check the 'jpg' file type filter and 'cc0' license filter, the link to 'all_content' type should only contain 'cc0' in the parameter, and discard the 'jpg' parameter.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
